units/schedule_patch.go

```diff
@@ -67,3 +85,34 @@ func SchedulePatch(ctx context.Context, env evergreen.Environment, patchId strin
 	}
 	return http.StatusOK, nil
 }
+
+// addPreviousSingleHostTaskGroupTasks checks all the tasks in a task group in the patch update
+// and if that task group is a single host task group, it schedules the previous tasks in the
+// task group.
+func addPreviousSingleHostTaskGroupTasks(patchUpdateReq *model.PatchUpdate, project *model.Project) {
+	taskGroupTasksToAddToVariant := map[string]string{}
+	for _, vt := range patchUpdateReq.VariantsTasks {
+		for _, t := range vt.Tasks {
+			tg := project.FindTaskGroupForTask(t)
+			if tg == nil || tg.MaxHosts != 1 {
+				continue
+			}
+			for _, tgt := range tg.Tasks {
+				if tgt == t {
+					break
+				}
+				taskGroupTasksToAddToVariant[tgt] = vt.Variant
```

Comment:
Maybe I'm missing something obvious, but since this is scheduling individual tasks in the task group, why doesn't this need a list of the previous task group tasks to add to the variant? For example, if we have a single host task group of task1, task2, and task3 and then schedule only task3, does this still schedule task1 and task2? Just reading through the logic really fast, it seems like it would only schedule task2.